src/test/kotlin/study/CalculatorValidTest.kt

```diff
@@ -0,0 +1,73 @@
+package study
+
+import calculator.ValidCheck
+import calculator.calculateOperationList
+import calculator.operation.Addition
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.EmptySource
+import org.junit.jupiter.params.provider.MethodSource
+import org.junit.jupiter.params.provider.NullSource
+import org.junit.jupiter.params.provider.ValueSource
+import java.util.stream.Stream
+
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CalculatorValidTest {
+
+    @Test
+    @DisplayName("문자열 split 배열 개수 유효성 검사")
+    fun splitValidCount() {
+        //given
+        val input = "2 + 3 * 4 /"
+
+        // when, then
+        assertThatThrownBy { ValidCheck.checkSplitArrayCount(input.split(" ")) }
+            .isInstanceOf(IllegalArgumentException::class.java)
+            .hasMessageContaining("숫자")
+            .hasMessageContaining("연산자")
+    }
+
+    @Test
+    @DisplayName("사칙연산 기호가 아닌 경우 검사")
+    fun splitValidOperation() {
+        //given
+        val input = "2 + 3 * 4 % 1"
+
+        // when, then
+        assertThatThrownBy { input.calculateOperationList() }
+            .isInstanceOf(IllegalArgumentException::class.java)
+            .hasMessageContaining("operator")
+    }
+
+    @ParameterizedTest
+    @NullSource
+    @EmptySource
+    @ValueSource(strings = ["", " ", "    "])
+    fun isEmpty_String_Exception(input: String?) {
+        //given, when, then
+        assertThatThrownBy { ValidCheck.checkEmptyString(input) }
+            .isInstanceOf(IllegalArgumentException::class.java)
+            .hasMessageContaining("null")
+            .hasMessageContaining("공백")
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideInputNumbers")
+    fun addition(a: Int, b: Int) {
+        //given, when, then
+        assertTrue(Addition().operation(a, b) == a + b)
+    }
```

Comment:
좋은 방법이에요! 하지만 저는 최대한 `@MethodSource`의 사용은 최대한 미루는 편이에요.
위와 같은 환경에서는 `@CsvSource`를 활용하면 좋겠네요!
그리고 `a`, `b`만 넘기는게 아닌, `result`의 값도 넘기는게 좋을것 같아요.
```suggestion
    @ParameterizedTest
    @CsvSource(value = ["2,3,5", "5,8,13"])
    fun addition(a: Int, b: Int, result: Int) {
        //given, when, then
        assertTrue(Addition().operation(a, b) == result)
    }
```